staker/block_validator.go

```diff
@@ -395,7 +395,7 @@ func GlobalStateToMsgCount(tracker InboxTrackerInterface, streamer TransactionSt
 		return false, 0, err
 	}
 	if res.BlockHash != gs.BlockHash || res.SendRoot != gs.SendRoot {
-		return false, 0, fmt.Errorf("%w: count %d hash %v expected %v, sendroot %v expected %v", ErrGlobalStateNotInChain, count, gs.BlockHash, res.BlockHash, gs.SendRoot, res.SendRoot)
+		return false, count, fmt.Errorf("%w: count %d hash %v expected %v, sendroot %v expected %v", ErrGlobalStateNotInChain, count, gs.BlockHash, res.BlockHash, gs.SendRoot, res.SendRoot)
```

Comment:
This change is to let the caller know the postition even it is wrong already. This helps us to submit and convince the wrong node